data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/utils/IPValidationcheck.java

```diff
@@ -31,7 +38,39 @@ public static boolean ipValidationcheck(String input) throws UnknownHostExceptio
      * @throws UnknownHostException UnknownHostException
      */
     public static boolean PrivateIPVersionCheck(String ipAddress) throws UnknownHostException {
-        //TODO : Checking for the IP is valid or not if IP is not a private IPV4 and IPV6
+        InetAddress address = InetAddress.getByName(ipAddress);
```

Comment:
Do we need this? Can we not use ``InetAddress.isSiteLocalAddress()`?

See http://docs.oracle.com/javase/1.5.0/docs/api/java/net/InetAddress.html#isSiteLocalAddress().

Reply:
Updated the code with InetAddress.isSiteLocalAddress.